.github/workflows/pre-release-workflow.yml

```diff
@@ -83,12 +88,20 @@ jobs:
 
       - name: Gradlew Pre-Release
         run: |
-          sed -i -r "s#version = ([^\']+)#version = ${{ github.event.inputs.current_version }}#" gradle.properties
+          sed -i -r "s#version = ([^\']+)#version = ${{ github.event.inputs.version_to_release }}#" gradle.properties
           git add gradle.properties
-          git commit -m "CI/CD: change version to be released: ${{ github.event.inputs.current_version }}"
+          git commit -m "CI/CD: change version to be released: ${{ github.event.inputs.version_to_release }}"
           git push
           ./gradlew clean publishReleasePublicationToSonatype closeSonatypeStagingRepository -Dorg.gradle.parallel=false --stacktrace
 
+      - name: Set next SDK version
```

Comment:
we shouldn't be setting the next version in pre-release-workflow. At least that's how I thought it.

pre-release-workflow will:
- create the staging repository in nexus
- upload the artefacts to it
- close repositories

at this stage, we are able to perform smoke testing. That is why it's called pre-release.

It is important that we do not update gradle.properties with next-version-SNAPSHOT in this step, because we need the actual "pre-released" version for the next workflow (release.yaml)

In release workflow we will:
- find the already created staging repository (this is why we don't need to update gradle.properties in previous step)
- make the actual release from sonatype to maven central

does it make sense? i wasn't able to test the full flow, but it should be working.

If so, we should make the same changes to swazzler

Reply:
So all the changes should be done to the release branch at `release/version`, so the release stage should be referring to that and not master.

The goal here is to take a snapshot of master and move it to another branch so that master can move along, assuming that everything after that goes into master is part of the next release. 

If there is something wrong in the release branch, we can fix it in master, and cherrypick the change and generate and test another release candidate. As long as we can do that in the release branch without needing to increment the patch version, we should be good.